src/RegistrationController.js

```diff
@@ -58,19 +62,65 @@ function (
     save: Okta.loc('registration.form.submit', 'login')
   });
 
+  var DEFAULT_CALLBACK_ERROR = 'We could not process your registration at this time. Please try again later';
+
   return BaseLoginController.extend({
     className: 'registration',
     initialize: function() {
       var clientId = this.options.settings.get('clientId');
       var registrationApi = this.options.settings.get('baseUrl')+'/api/v1/registration/'+clientId;
 
-      var Schema = RegistrationSchema.extend({
-        url: registrationApi+'/form'
-      });
-      
-      var schema = new Schema();
+      var doPostSubmit = _.bind(function () {
+        if (this.model.get('activationToken')) {
+          var authClient = this.model.appState.settings.authClient;
+          authClient.signIn({
+            token: this.model.get('activationToken')
+          })
+          .then(_.bind(function(transaction) {
```

Comment:
Why do we need to manually call routeAfterAuthStatusChange? Shouldn't it automatically just hook into the existing widget logic?

Reply:
Not a 100% sure about that, need to check . This was existing code that just got moved around.